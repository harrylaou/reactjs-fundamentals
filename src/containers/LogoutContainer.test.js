import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import ConnectedLogoutContainer, {LogoutContainer} from './LogoutContainer'
import getMuiTheme from 'material-ui/styles/getMuiTheme'

import FakeConfigureStore from 'redux-mock-store'
import ActualConfigureStore from '../store'
import {Provider} from 'react-redux'

import injectTapEventPlugin from 'react-tap-event-plugin'
injectTapEventPlugin()

describe('<LogoutContainer />', () => {
  it('should render with default props', () => {
    const logout = jest.fn()
    const push = jest.fn()
    const props = {
      params: { username: '123' },
      color: '',
      router: {push},
      logout
    }

    const wrapper = shallow(
      <LogoutContainer {...props} />
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()

    wrapper.instance().logoutUser()

    expect(logout).toBeCalled()
    expect(push).toBeCalledWith('/login')
  })


  const initialState = {
  }
  const mockStore = FakeConfigureStore()
  const logout = jest.fn()
  const push = () => {}
  const replace = () => {}
  const go = () => {}
  const goBack = () => {}
  const goForward = () => {}
  const setRouteLeaveHook = () => {}
  const isActive = () => {}

  const props = {
    router: {
      push,
      replace,
      go,
      goBack,
      goForward,
      setRouteLeaveHook,
      isActive,
      setRouteLeaveHook
    },
    color: '',
    logout
  }
  let store

  describe('<ConnectedLogoutContainer />', () => {
    it('should render connected component', () => {
      store = mockStore(initialState)
      const container = shallow(
        <ConnectedLogoutContainer
          {...props}
          store={store}
        />
      )

      expect(shallowToJson(container)).toMatchSnapshot()
    })
  })

  describe('Integration test: Mount + Wrapping Provider', () => {
    let wrapper

    beforeEach(()=>{
      store = mockStore(initialState)

      wrapper = mount(
        <Provider store={store} >
          <ConnectedLogoutContainer
            {...props}
          />
        </Provider>,
        {
          context: {
            muiTheme: getMuiTheme()
          },
          childContextTypes:{
            muiTheme: React.PropTypes.object.isRequired,
          }
        }
      )
    })

    it('should mount and map connect connected component', () => {
      expect(wrapper.find(ConnectedLogoutContainer).length).toEqual(1)
    })
  })

  describe('Integration test: Mount + Wrapping Provider + Actual Store + Reducers', () => {
    let wrapper

    beforeEach(()=>{
      store = ActualConfigureStore(initialState)

      wrapper = mount(
        <Provider store={store} >
          <ConnectedLogoutContainer
            {...props}
          />
        </Provider>,
        {
          context: {
            muiTheme: getMuiTheme()
          },
          childContextTypes:{
            muiTheme: React.PropTypes.object.isRequired,
          }
        }
      )
    })

    it('should mount and map connect connected component', () => {
      expect(wrapper.find(ConnectedLogoutContainer).length).toEqual(1)
    })
  })
})