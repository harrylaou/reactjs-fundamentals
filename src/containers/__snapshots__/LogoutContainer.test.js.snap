// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutContainer /> <ConnectedLogoutContainer /> should render connected component 1`] = `
<Connect(LogoutContainer)
  color=""
  logout={[Function]}
  router={
    Object {
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "isActive": [Function],
      "push": [Function],
      "replace": [Function],
      "setRouteLeaveHook": [Function],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`<LogoutContainer /> should render with default props 1`] = `
<Logout
  color=""
  logout={[Function]}
  logoutUser={[Function]}
  params={
    Object {
      "username": "123",
    }
  }
  router={
    Object {
      "push": [Function],
    }
  }
/>
`;
